### PR TITLE
Add `flake8-simplify` to linters

### PR DESCRIPTION
**Pull request by namurphy**
This PR adds [`flake8-simplify`](https://github.com/MartinThoma/flake8-simplify) as a flake8 extension, and implements most of the suggested changes.

### CODE REVIEW

File: plasmapy/formulary/parameters.py [hunk 1]
```diff
@@ -54,17 +56,11 @@
...
-    try:
+    with contextlib.suppress(ValueError):  # ValueError if name was not in __aliases__
         __aliases__.remove(name)
-    except ValueError:
-        # name was not in __aliases__
-        pass
 
-    try:
+    with contextlib.suppress(ValueError):  # ValueError if name was not in __all__
         __all__.remove(name)
```
Comment (StanczakDominik) [inline plasmapy/formulary/parameters.py:63]: These could be the same block.

**Review by StanczakDominik — APPROVED** [review]
LGTM!

File: plasmapy/formulary/parameters.py (hunk 1 above)
Comment (namurphy) [reply plasmapy/formulary/parameters.py:63]: I had wondered the same thing, but if there's an exception raised in the first line of the [`contextlib.suppress`](https://docs.python.org/3/library/contextlib.html#contextlib.suppress) block, the second line won't be run.  That would have been nice, though.

**Comment by namurphy** [comment]
Many thanks for the quick review!

File: plasmapy/formulary/parameters.py (hunk 1 above)
Comment (StanczakDominik) [reply plasmapy/formulary/parameters.py:63]: Oh, huh, didn't know that! Then there really is no difference between the try block and the suppress call.

**Pull request merged**